sssom/sssom_datamodel.py

```diff
@@ -299,7 +299,9 @@ def __post_init__(self, *_: List[str], **kwargs: Dict[str, Any]):
         if self.mapping_date is not None and not isinstance(self.mapping_date, str):
             self.mapping_date = str(self.mapping_date)
 
-        if self.confidence is not None and not isinstance(self.confidence, float):
+        if self.confidence == '' or self.confidence is None:
```

Comment:
Unfortunately sssom_datamodel.py cannot be manually editted, the other comment..